### DISCUSSION TITLE
A busy 48 hours!...

### DISCUSSION THREAD

**Discussion by WaynePhillipsEA** [twinbasic/twinbasic#168] [General]
Wow, just looking back, and after the 2-week lull, I've closed 15 issues over the last 48 hours.   Just in case you missed them, here's the changelog:
https://marketplace.visualstudio.com/items/twinbasic.twinbasic/changelog

In other news, I thought it worth mentioning here that I've now decided on the approach for GUI support.  twinBASIC will eventually offer two types of GUI forms.  One for Windows-only, which is fully backwards compatible with VB6, and a second type of form that supports cross-platform but is not 100% backwards compatible.  I'll be making every effort to make the cross-platform forms 'feel' like VB6 forms as much as possible, and the form designer will be shared between them both.

Have a nice weekend folks!

**Comment by bclothier** [comment]
Thrilled - I'm sure the lull will be worth it. 

For my education, what are the issues that prevents full compatibility between the traditional VB form architecture and cross platform?

**Reply by WaynePhillipsEA** [reply]
For cross-platform forms, controls will be fully rendered by tB, so we're not relying on any Windows standard controls at all.  Windows handles like HWNDs, HDCs etc won't be available, but we will instead provide OOP based drawing capabilities that will be identical across all supported platforms.   Things like ActiveX controls will also not be supported on the cross-platform forms, but of course will be supported on the VB6 compatible forms.

**Comment by Kr00l** [comment]
Interesting that GUI comes next. Wish you good success/progress with it.
I can't wait to see my VBCCR running and working in tB. :-)
This will become a real boost for tB as this signals "it works, let's migrate".
